conda/core/envs_manager.py

```diff
@@ -128,3 +131,46 @@ def _rewrite_environments_txt(environments_txt_file, prefixes):
     except EnvironmentError as e:
         log.info("File not cleaned: %s", environments_txt_file)
         log.debug('%r', e, exc_info=True)
+
+
+def _get_environment_state_file(prefix):
+    env_vars_file = join(prefix, PREFIX_SATE_FILE)
+    if exists(env_vars_file):
+        with open(env_vars_file, 'r') as f:
+            prefix_state = json.loads(f.read(), object_pairs_hook=OrderedDict)
+    else:
+        prefix_state = {}
+    return prefix_state
+
+
+def _write_environment_state_file(prefix, state):
+    env_vars_file = join(prefix, PREFIX_SATE_FILE)
+    with open(env_vars_file, 'w') as f:
+        f.write(json.dumps(state, ensure_ascii=False, default=lambda x: x.__dict__))
+
+
+def get_environment_env_vars(prefix):
+    prefix_state = _get_environment_state_file(prefix)
+    env_vars = OrderedDict(prefix_state.get('env_vars', {}))
+    return env_vars
+
+
+def set_environment_env_vars(prefix, env_vars):
+    env_state_file = _get_environment_state_file(prefix)
+    current_env_vars = env_state_file.get('env_vars')
+    if current_env_vars:
+        current_env_vars.update(env_vars)
+    else:
+        env_state_file['env_vars'] = env_vars
+    _write_environment_state_file(prefix, env_state_file)
+    return env_state_file['env_vars']
+
+
+def unset_environment_env_vars(prefix, env_vars):
+    env_state_file = _get_environment_state_file(prefix)
+    current_env_vars = env_state_file.get('env_vars')
+    if current_env_vars:
+        for env_var in env_vars:
+            current_env_vars.pop(env_var)
+        _write_environment_state_file(prefix, env_state_file)
+    return env_state_file['env_vars']
```

Comment:
Consider moving the changes you've made here to `prefix_data.py`. The `PrefixData` class is what holds all the state for a particular prefix.  The code in `envs_manager.py` is really a level above a single prefix, meant to inspect and interact with all of the prefixes on a given system.